hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1008,6 +1012,10 @@ func (r *NodePoolReconciler) setAllMachinesReadyCondition(nodePool *hyperv1.Node
 		status = corev1.ConditionFalse
 		reason = hyperv1.NodePoolNotFoundReason
 		message = "No Machines are created"
+		if nodePool.Spec.Replicas != nil && *nodePool.Spec.Replicas == 0 {
+			reason = hyperv1.AsExpectedReason
+			message = "NodePool replicas set to nil"
```

Comment:
`replicas` could also be set to 0, right? not `nil`.

Maybe "NodePool set to no replicas" instead?

Reply:
I thought nil also meant zero, but I see where the confusion could be.